druid/src/widget/list.rs

```diff
@@ -247,30 +280,61 @@ impl<C: Data, T: ListIter<C>> Widget<T> for List<C> {
                     return;
                 }
             };
-            let child_bc = BoxConstraints::new(
-                Size::new(bc.min().width, 0.0),
-                Size::new(bc.max().width, std::f64::INFINITY),
-            );
-            let child_size = child.layout(ctx, &child_bc, child_data, env);
-            let rect = Rect::from_origin_size(Point::new(0.0, y), child_size);
+            let child_bc = match layout_direction {
+                ListDirection::Vertical => BoxConstraints::new(
+                    Size::new(bc.min().width, 0.0),
+                    Size::new(bc.max().width, std::f64::INFINITY),
+                ),
+                ListDirection::Horizontal => BoxConstraints::new(
+                    Size::new(0.0, bc.min().height),
+                    Size::new(std::f64::INFINITY, bc.max().height),
+                ),
+            };
+            let child_size = match layout_direction {
+                ListDirection::Vertical => Size::new(
+                    my_size.width,
+                    child.layout(ctx, &child_bc, child_data, env).height,
+                ),
+                ListDirection::Horizontal => Size::new(
+                    child.layout(ctx, &child_bc, child_data, env).width,
+                    my_size.height,
+                ),
+            };
+            let rect = match layout_direction {
+                ListDirection::Vertical => {
+                    Rect::from_origin_size(Point::new(0.0, height), child_size)
+                }
+
+                ListDirection::Horizontal => {
+                    Rect::from_origin_size(Point::new(width, 0.0), child_size)
+                }
+            };
             child.set_layout_rect(ctx, child_data, env, rect);
             paint_rect = paint_rect.union(child.paint_rect());
-            width = width.max(child_size.width);
-            y += child_size.height;
+
+            match layout_direction {
+                ListDirection::Vertical => height += child_size.height,
+                ListDirection::Horizontal => width += child_size.width,
+            }
         });
 
-        let my_size = bc.constrain(Size::new(width, y));
+        self.scroll_component.content_size = Size::new(width, height);
         let insets = paint_rect - Rect::ZERO.with_size(my_size);
```

Comment:
```suggestion
        let insets = paint_rect - my_size.to_rect();
```

Reply:
I hadn't modified this line but this seems a reasonable minor change which shouldn't be massively out of scope to include